http/controllers/UserController.ts

```diff
@@ -51,19 +96,61 @@ export class UserController {
     ]
 
     ctx.body = {
-      user: updatedUser,
+      user: {
+        firstName: user.firstName,
+        lastName: user.lastName,
+        username: user.username,
+        email: user.email,
+      },
     }
   }
 
   delete = async (ctx: IRouterContext) => {
     const userId: number = parseInt(ctx.params.id)
     const userIndex: number = this.users.findIndex(user => user.id === userId)
 
+    if (userIndex < 0) {
+      ctx.status = 404
+      ctx.body = {
+        error: 'User does not exist',
+      }
+
+      return
+    }
+
     this.users = [
       ...this.users.slice(0, userIndex),
       ...this.users.slice(userIndex + 1),
     ]
 
     ctx.status = 201
   }
+
+  login = async (ctx: IRouterContext) => {
+    const reqUser: any = ctx.request.body
+    const user: User = this.users.find(user => user.email === reqUser.email)
+    const passwordVerified: boolean = await bcrypt.compareSync(
+      reqUser.password,
+      user.password
```

Comment:
if user is returned as null from `.find` would this throw an error? maybe do the `401` or `404` before comparing the passwords?

Also, you are using compareSync which is synchronous, meaning you don't need the `await` you could switch this to the async version